ros/angel_system_nodes/angel_system_nodes/task_monitor.py

```diff
@@ -346,7 +351,8 @@ def task_timer_thread(self):
             self._timer_active = False
 
         # Advance to the next state
-        self._task.trigger(self._task.state)
+        with self._task_lock:
+            self._task.trigger(self._task.state)
```

Comment:
This might not do what we expect if a manual override keypress happens during a timer (double forward). We may want to record the current state when this method starts, and either check during/after the loop that we are still in that state before advancing (or let the trigger error with a MachineError because we are no longer in the state to progress from) and return early.

Reply:
Ok, I added a check to make sure the state doesn't change during the  timer and tested it. Also, I moved the check for if a timer is active during the ActivityDetection callback to the beginning of that function so that it doesn't try to switch activities while a timer is going.